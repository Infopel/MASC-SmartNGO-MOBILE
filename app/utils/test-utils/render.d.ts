export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
