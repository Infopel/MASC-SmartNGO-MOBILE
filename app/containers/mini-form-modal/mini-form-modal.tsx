import { FormBody } from "components/form-body/form-body"
import { Heading } from "components/heading/heading"
import { Actionsheet, AlertDialog, HStack } from "native-base"
import { PrimaryNavigatorParamList } from "navigators/navigators.d"
import * as React from "react"
import { useFormStore } from "store/form/form.store"

export type MiniFormModalProps = PrimaryNavigatorParamList['form'] & {
  isOpen: boolean
  onClose: () => void
}

/**
 * Describe your component here
 */
export function MiniFormModalContainer(props: MiniFormModalProps) {
  const { formId, contextId, parentContextId, isOpen = false, onClose } = props

  const { fields, requestForm, isFirstPage, nextPage, isLastPage, prevPage, clear, submitForm, pageInfo } = useFormStore(({ fields, requestForm, page, nextPage, isLastPage, isFirstPage, prevPage, clear, submitForm }) => ({ submitForm, fields, clear, nextPage, isFirstPage: isFirstPage(), isLastPage: isLastPage(), prevPage, pageInfo: page(), requestForm }))

  React.useEffect(() => {
    requestForm(formId, contextId, parentContextId)
    return () => clear()
  }, [formId, contextId,parentContextId])

  const handleOnClose = () => {
    clear()
    onClose()
  }


  return (
    <Actionsheet isOpen={isOpen} onClose={handleOnClose}>
      <Actionsheet.Content h='xl'>
        <HStack mb='4' justifyContent='space-between' space='0'>
          <Heading
            w='full'
            numberOfLines={1}
            fontWeight='light'
            m='3'
            tx={pageInfo?.title}
            textBreakStrategy='simple' />
          <AlertDialog.CloseButton onPress={handleOnClose} />
        </HStack>
        <FormBody fields={fields()} {...{ nextPage, prevPage, isLastPage, isFirstPage, submitForm }} />
      </Actionsheet.Content>
    </Actionsheet>
  )
}
