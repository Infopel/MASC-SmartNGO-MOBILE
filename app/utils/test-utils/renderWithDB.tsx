jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  