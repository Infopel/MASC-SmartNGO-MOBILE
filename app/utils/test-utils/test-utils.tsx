import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

