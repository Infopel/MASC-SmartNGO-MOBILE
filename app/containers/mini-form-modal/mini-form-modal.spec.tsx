import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { MiniFormModalContainer } from "./mini-form-modal"

describe("<MiniFormModalContainer>",()=>{
 it("render", async function () {
    const {toJSON} = render(<MiniFormModalContainer></MiniFormModalContainer>)
    
    expect(toJSON()).toMatchSnapshot()
  })
})
  