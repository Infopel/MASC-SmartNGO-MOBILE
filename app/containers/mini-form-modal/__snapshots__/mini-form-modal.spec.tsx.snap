// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MiniFormModalContainer> render 1`] = `
<View
  style={
    Object {
      "padding": 16,
    }
  }
  testID="miniFormModalContainer"
>
  <Text
    dataSet={Object {}}
    style={
      Array [
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        },
        Object {
          "color": "#FBA928",
          "fontFamily": "Helvetica",
          "fontSize": 14,
        },
      ]
    }
    testID="miniFormModalContainerTestId"
  >
    Hello
  </Text>
</View>
`;
